src/Microsoft.ML.Data/Transforms/Normalizer.cs

```diff
@@ -554,7 +576,66 @@ private bool SaveAsOnnxCore(OnnxContext ctx, int iinfo, ColumnInfo info, string
 
                 return false;
             }
+        }
+
+        /// <summary>
+        /// An interface implemented by items of <see cref="ColumnFunctions"/> corresponding to the
+        /// <see cref="NormalizeTransform.AffineColumnFunction"/> items.
+        /// </summary>
+        internal interface IAffineData<TData>
+        {
+            /// <summary>
+            /// The scales. In the scalar case, this is a single value. In the vector case this is of length equal
+            /// to the number of slots. Function is <c>(input - offset) * scale</c>.
+            /// </summary>
+            TData Scale { get; }
+
+            /// <summary>
+            /// The offsets. In the scalar case, this is a single value. In the vector case this is of length equal
+            /// to the number of slots, or of length zero if all the offsets are zero.
+            /// </summary>
+            TData Offset { get; }
+        }
 
+        /// <summary>
+        /// An interface implemented by items of <see cref="ColumnFunctions"/> corresponding to the
+        /// <see cref="NormalizeTransform.CdfColumnFunction"/> items. The function is the value of the
+        /// cumulative density function of the normal distribution parameterized with mean <see cref="Mean"/>
+        /// and standard deviation <see cref="Stddev"/>.
+        /// </summary>
+        internal interface ICdfData<TData>
+        {
+            /// <summary>
+            /// The mean(s). In the scalar case, this is a single value. In the vector case this is of length equal
+            /// to the number of slots.
+            /// </summary>
+            TData Mean { get; }
+
+            /// <summary>
+            /// The standard deviation(s). In the scalar case, this is a single value. In the vector case this is of
+            /// length equal to the number of slots.
+            /// </summary>
+            TData Stddev { get; }
+
+            /// <summary>
+            /// Whether the input should first be considered to be logged, that is, whether this was configured
```

Comment:
>logged [](start = 67, length = 6)

I am not sure, is 'logged' a term from timber industry? Maybe 'on log scale' ?

Reply:
That's right. We live in the pacific northwest after all.

Seriously, yes I agree, also the "considered" as is "considered to be on the log scale" also is not a great description, since what actually happens is the code applies a logarithm, period. What can I say. :) So a simpler time.

---
In reply to: [215736736](https://github.com/dotnet/machinelearning/pull/843#discussion_r215736736) [](ancestors = 215736736)